clang/test/CIR/CodeGen/array.cpp

```diff
@@ -46,3 +46,18 @@ void local_stringlit() {
 // CHECK-NEXT:  %1 = cir.get_global @".str" : cir.ptr <!cir.array<!s8i x 8>>
 // CHECK-NEXT:  %2 = cir.cast(array_to_ptrdecay, %1 : !cir.ptr<!cir.array<!s8i x 8>>), !cir.ptr<!s8i>
 // CHECK-NEXT:  cir.store %2, %0 : !cir.ptr<!s8i>, cir.ptr <!cir.ptr<!s8i>>
+
+int multidim(int i, int j) {
+  int arr[2][2];
+  return arr[i][j];
+}
+
+// CHECK: %3 = cir.alloca !cir.array<!cir.array<!s32i x 2> x 2>, cir.ptr <!cir.array<!cir.array<!s32i x 2> x 2>>
+// Stride first dimension (stride = 2)
+// CHECK: %4 = cir.load %{{.+}} : cir.ptr <!s32i>, !s32i
+// CHECK: %5 = cir.cast(array_to_ptrdecay, %3 : !cir.ptr<!cir.array<!cir.array<!s32i x 2> x 2>>), !cir.ptr<!cir.array<!s32i x 2>>
+// CHECK: %6 = cir.ptr_stride(%5 : !cir.ptr<!cir.array<!s32i x 2>>, %4 : !s32i), !cir.ptr<!cir.array<!s32i x 2>>
```

Comment:
I wonder whether we want to raise multi-dim access looking into the future, as to make it easier to lower to other dialects that operate well on them, just wonder if you had similar thoughts

Reply:
I do! It might be especially useful for polyhedral stuff.

Though, I envisioned this high-level representation not in CIR it self, but in something like a future high-level CIR dialect. This way we could go `HCIR -> CIR` or `HCIR -> MLIR -> CIR`.